specification/cognitiveservices/AnomalyDetector/cadl-project.yaml

```diff
@@ -12,7 +12,6 @@ emitters:
   # Uncomment this line and add "@azure-tools/cadl-csharp" to your package.json to generate C# code
   "@azure-tools/cadl-csharp":
     save-inputs: false
-    clear-output-folder: true
```

Comment:
'clear-output-folder' is needed. Think when you update the client name, we need to delete previous generated <previousName>Client.cs. If you does not clear the folder, the <previousName>Client.cs will remain there which is not correct. Same way as delete/rename model.

I think the previous code is the one already existed in the azure-sdk-for-net, Not the one iteratively generated by customer. We need to figure out way to find out the right base coded we need to import.

Reply:
I don't think changing name is allowed. It is breaking change.

Reply:
If user really wants clear the output folder, they could set it to true. However, I don't think anomaly detector wants to do that.

Reply:
Changing the name is a breaking change this is true, but it is not an excuse that the generator could generate invalid code if that really happens. The generator should always generate valid code, if it cannot, it should give proper error message to describe what error happens.
Therefore I disagree to remove the `clear-output-folder` here.
I believe the root cause we have to remove this for this RP is because this flag is handled by the emitter and the generator needs the previous generated code - the generator cannot read it if we removed those in the emitter. Is this the case?
If this is true, we should verify and fix that in the emitter instead of removing this flag here.

Reply:
> but it is not an excuse that the generator could generate invalid code if that really happens.

What invalid code you mean?

> Therefore I disagree to remove the clear-output-folder here.

Sounds like you disagree to remove it for all the services. Then why we have this parameter exposed?

> I believe the root cause we have to remove this for this RP is because this flag is handled by the emitter and the generator needs the previous generated code

No. It is because if we remove it, codegen will take it as there is no existing code, so codegen will generate overloads for convenience methods instead of renaming them.

Reply:
First the existing code is the one existing in `azure-sdk-for-net`, not equal to the `previously generated one`.

And `clear-output-folder` is needed, otherwise it will cause issue when generate it next time as we mentioned above.

To resolve the issue of `import existing code`, we need to move `delete` action into codegen maybe, and delete them after we import the `existing code`, and we should be careful about the `existing code` we selected to import.